sdk/src/Services/S3/Custom/Transfer/Internal/MultipartUploadCommand.cs

```diff
@@ -230,7 +247,43 @@ private UploadPartRequest ConstructUploadPartRequest(int partNumber, long filePo
             return uploadRequest;
         }
 
+        private UploadPartRequest ConstructUploadPartRequestForNonSeekableStream(Stream inputStream, int partNumber, long partSize, bool isLastPart, InitiateMultipartUploadResponse initiateResponse)
+        {
+            UploadPartRequest uploadPartRequest = new UploadPartRequest()
+            {
+                BucketName = this._fileTransporterRequest.BucketName,
+                Key = this._fileTransporterRequest.Key,
+                UploadId = initiateResponse.UploadId,
+                InputStream = inputStream,
+                PartSize = partSize,
+                PartNumber = partNumber,
+                IsLastPart = isLastPart,
+                ServerSideEncryptionCustomerMethod = this._fileTransporterRequest.ServerSideEncryptionCustomerMethod,
+                ServerSideEncryptionCustomerProvidedKey = this._fileTransporterRequest.ServerSideEncryptionCustomerProvidedKey,
+                ServerSideEncryptionCustomerProvidedKeyMD5 = this._fileTransporterRequest.ServerSideEncryptionCustomerProvidedKeyMD5,
+#if (BCL && !BCL45)
+                                    Timeout = ClientConfig.GetTimeoutValue(this._config.DefaultTimeout, this._fileTransporterRequest.Timeout),
+#endif
+                DisableMD5Stream = this._fileTransporterRequest.DisableMD5Stream,
+                DisablePayloadSigning = this._fileTransporterRequest.DisablePayloadSigning,
+                ChecksumAlgorithm = this._fileTransporterRequest.ChecksumAlgorithm,
+                CalculateContentMD5Header = this._fileTransporterRequest.CalculateContentMD5Header
+            };
+
+            // If the InitiateMultipartUploadResponse indicates that this upload is using KMS, force SigV4 for each UploadPart request
```

Comment:
Minor: Will defer to norm here. Do we need to add the progress Handler here like we do for regular streams?